InstanceExport/PowerShell/InstanceExport.ps1

```diff
@@ -187,150 +189,162 @@ function Export {
-            Exit 1
-        }
-    
-        try {
-            CreateDirectoryIfDoesNotExist -directoryPath $exportFilePath -currentDate $currentDate
-        }
-        catch {
-            $msg = "Error trying to create $directoryPath directory"
-            $displayMsg = "An error occurred when trying to create a new directory"
-            Log -msg $msg -displayMsg $displayMsg -logLevel "error" -currentDate $currentDate
-            Exit 1
-        }
-
-        try {
-            $accessTokenResponseModel = Login
-        }
-        catch {
-            Log -msg "An error occurred" -displayMsg "$($_.Exception.Response.StatusCode.value__): An error occurred when logging in" -logLevel "error" -currentDate $currentDate
-            Log -msg "Authorization error for Instance Export" -logLevel "error" -currentDate $currentDate
-            Log -msg "StatusCode: $($_.Exception.Response.StatusCode.value__)" -logLevel "error" -currentDate $currentDate
-            Log -msg "Url: api/login" -logLevel "error" -currentDate $currentDate
-            Log -msg "StatusDescription: $($_.Exception.Response.StatusDescription)" -logLevel "error" -currentDate $currentDate
-        }
-
-        if ($accessTokenResponseModel) {
-
-            $accessToken = $accessTokenResponseModel.access_token
-
-            $header = @{
-                "authorization" = "Bearer $accessToken"
-            }
-
-            $currentCategory = "";
-            $i = 0;
-            for (; $i -lt $manifestJson.Entries.Length; $i = $i + 1) {
-                $entry = $manifestJson.Entries[$i]
-                if ($currentCategory -ne $entry.Category) {
-                    $currentCategory = $entry.Category;
-                    $message = "Starting extraction of $($entry.Category) category."
-                    Log -msg $message -displayMsg $message -logLevel "displayInfo" -currentDate $currentDate
+	
+    $manifestFilePathExists = Test-Path -Path $manifestFilePath
+	
+    if ($manifestFilePathExists) {
+        $manifestFilePath = $manifestFilePath.TrimEnd('\')  # Trim any trailing backslash if it exists
+		
+        $manifesJsonFiles = Get-ChildItem -Path "$manifestFilePath\*" -Include "*.json"
+		
+        foreach ($manifestFile in $manifesJsonFiles) {
+			
+            $manifestFileExists = Test-Path -Path $manifestFile
+			
+            $instance = ($manifestFile.BaseName -split ' ')[0]
+			
+            if ($manifestFileExists) {
+                $manifestJson = Get-Content -Raw -Path $manifestFile | ConvertFrom-Json
+                if ($manifestJson.CreateDate) {
+                    $currentDate = Get-Date $manifestJson.CreateDate -Format "MM_dd_yyyy_HH_mm_ss"
+                }
+                else {
+                    $msg = "Error in reading manifest file"
+                    $displayMsg = "An error occurred when reading manifest file"
+                    Log -msg $msg -displayMsg $displayMsg -logLevel "error"  -currentDate $currentDate -instance $instance
```

Comment:
If there's an error reading the manifest file, this `Log` call will crash because the instance folder does not exist
```
Add-Content : Could not find a part of the path
'C:\Code\DevResultsTools\InstanceExport\PowerShell\export\foo\foo_InstanceExport_04_13_2025_12_50_58_log.txt'.
At C:\Code\DevResultsTools\InstanceExport\PowerShell\InstanceExport.ps1:96 char:5
+     Add-Content "$($exportFilePath)/$instance/$($instance)_InstanceEx ...
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (C:\Code\DevResu...2_50_58_log.txt:String) [Add-Content], DirectoryNotFoundException
    + FullyQualifiedErrorId : GetContentWriterDirectoryNotFoundError,Microsoft.PowerShell.Commands.AddContentCommand
```